### PR TITLE
Feature/api call event

### PR DESCRIPTION
**Pull request by chaoexit**

### CODE REVIEW

File: client/events/index.js [hunk 1]
```diff
@@ -48,10 +55,10 @@ function handleEvent (type, event) {
   data.API_KEY_PUBLIC = this.API_KEY_PUBLIC
 
   if (!(
-    ((type === 'mousemove' || type === 'resize') && !requestIsNotInDelay(type)) ||
+    (isMouseMoveAndReize(type) && !requestIsNotInDelay(type)) ||
     (type === 'APICall' && !isCallToProductEndPoint(data.url.hostname))
-  )) {
-    callFetch(fetch, type, data)
+  ) && !hasError) {
+    callFetch(fetch, type, data, this)
```
Comment (0xchimz) [inline client/events/index.js]: - refactor code 
`callFetch.call(this, type, data)`

File: client/util/fetch.js [hunk 2]
```diff
@@ -77,6 +86,8 @@ class BrowserFetch {
   }
 }
 
+let hasError = false
```
Comment (0xchimz) [inline client/util/fetch.js]: hasError in BrowserFetch is never used

File: client/events/index.js [hunk 3]
```diff
@@ -18,13 +18,20 @@ const propertiesObject = {
   scroll: scrollProperties
 }
 
+let hasError = false
 let prevTime = {
   mousemove: new Date(0),
   resize: new Date(0)
 }
 
 function callFetch (fetch, type, data) {
```
Comment (0xchimz) [inline client/events/index.js]: bind callFetch method for calling fetch function with this.fetch

**Review by 0xchimz — CHANGES_REQUESTED** [review]

**Comment by chaoexit** [comment]
Modified as reviewed, please check @chinclubi

**Review by 0xchimz — APPROVED** [review]

**Pull request merged**